nvflare/lighter/ctx.py

```diff
@@ -103,8 +103,20 @@ def yaml_load_template_section(self, section_key: str):
     def json_load_template_section(self, section_key: str):
         return json.loads(self.get_template_section(section_key))
 
-    def build_from_template(self, dest_dir: str, temp_section: str, file_name, replacement=None, mode="t", exe=False):
+    def build_from_template(
+        self,
+        dest_dir: str,
+        temp_section: str,
+        file_name,
+        replacement=None,
+        mode="t",
+        exe=False,
+        content_modify_cb=None,
+        **cb_kwargs,
+    ):
         section = self.get_template_section(temp_section)
         if replacement:
             section = utils.sh_replace(section, replacement)
+        if content_modify_cb:
+            section = content_modify_cb(section, **cb_kwargs)
```

Comment:
suggest add a docstring explaining that "content_modify_cb" will need to have the first argument being section